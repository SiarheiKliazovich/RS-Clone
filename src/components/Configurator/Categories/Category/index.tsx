import { FunctionComponent } from "react";
import { CategoryType } from "../../../../types/configurator";
import { nightTheme } from "../../../../data/constants";
import "./category.scss";

const Category: FunctionComponent<CategoryType> = ({
  tie,
  setType,
  accentColor,
  isNavbarNightMode,
}: CategoryType) => {
  const backgroundColor = nightTheme.background.element;

  return (
    <li className="category__list_item">
      <div className="category-blok">
        <img className="category-img" src={tie.image} alt="img-ties" />

        <button
          style={{
            backgroundColor: isNavbarNightMode
              ? backgroundColor
              : accentColor.static,
          }}
          className="category-name"
          onClick={() => setType(tie.type)}
        >
          {tie.categoryName}
        </button>
      </div>
    </li>
  );
};

export default Category;
