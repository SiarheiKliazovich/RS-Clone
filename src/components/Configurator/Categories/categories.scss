@import "../../../sass/variables";
@import "../../../sass/mixins";

.category__list {
  list-style: none;
  display: flex;
  text-align: center;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0;
  padding: 0;
  gap: 20px 10px;

  @include media-small-mobile {
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
}
