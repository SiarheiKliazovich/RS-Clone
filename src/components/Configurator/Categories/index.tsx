import Category from "../Categories/Category";
import { FunctionComponent } from "react";
import { CategoriesType } from "../../../types/configurator";
import { ICategory } from "../../../interfaces/configurator";

const Categories: FunctionComponent<CategoriesType> = ({
  tieList,
  setType,
}: CategoriesType) => {
  return (
    <ul className="category__list">
      {tieList?.data?.map((tie: ICategory) => (
        <Category tie={tie} setType={setType} key={tie.type} />
      ))}
    </ul>
  );
};

export default Categories;
