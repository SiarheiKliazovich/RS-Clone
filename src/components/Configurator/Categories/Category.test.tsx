import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Categories from ".";

describe("Categories", () => {
  const tieList = {
    lang: "ru",
    data: [
      {
        categoryName: "Центральный узор",
        image: "https://i.ibb.co/t4RqPjL/centered.jpg",
        id: 3,
        type: "centred",
        price: "12",
        subCategories: [
          {
            name: "CENT 507",
            image:
              "https://api.kitbuilder.co.uk/svg/defaultimage/6503301?width=300&height=300&view=0&cachedSvgVersion=06ca279a-645b-4822-a5d8-45c1ed02d51a&key=undefined&distributorId=76104207",
            id: 32,
            type: "CENT507",
          },
          {
            name: "CENT 512",
            image:
              "https://api.kitbuilder.co.uk/svg/defaultimage/6503302?width=300&height=300&view=0&cachedSvgVersion=ba87498e-52b4-4247-90bd-417093d6b2a3&key=undefined&distributorId=76104207",

            id: 33,
            type: "CENT512",
          },
          {
            name: "CENT 516",
            image:
              "https://api.kitbuilder.co.uk/svg/defaultimage/6503303?width=300&height=300&view=0&cachedSvgVersion=a62cdbdd-1d8e-4b28-8f7f-ceafdbc67272&key=undefined&distributorId=76104207",

            id: 34,
            type: "CENT516",
          },
          {
            name: "CENT 517",
            image:
              "https://api.kitbuilder.co.uk/svg/defaultimage/6503304?width=300&height=300&view=0&cachedSvgVersion=bcbeb6d1-0b72-4e24-b30c-10343ac35324&key=undefined&distributorId=76104207",

            id: 35,
            type: "CENT517",
          },
          {
            name: "CENT 520",
            image:
              "https://api.kitbuilder.co.uk/svg/defaultimage/6503305?width=300&height=300&view=0&cachedSvgVersion=053231d3-ab56-475b-9406-4b661fa7a353&key=undefined&distributorId=76104207",

            id: 36,
            type: "CENT520",
          },
        ],
      },
    ],
  };

  const accentColor = {
    static: "#000",
    hover: "#fff",
  };

  const isNavbarNightMode = false;

  test("renders correctly", () => {
    const { container } = render(
      <Router>
        <Categories
          tieList={tieList}
          accentColor={accentColor}
          isNavbarNightMode={isNavbarNightMode}
        />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
