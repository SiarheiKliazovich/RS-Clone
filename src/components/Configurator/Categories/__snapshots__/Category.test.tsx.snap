// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<ul
  class="category__list"
>
  <li
    class="category__list_item"
  >
    <div
      class="category-blok"
    >
      <img
        alt="img-ties"
        class="category-img"
        src="https://i.ibb.co/t4RqPjL/centered.jpg"
      />
      <a
        class="category-name"
        href="/configurator/centred"
        style="background-color: rgb(0, 0, 0);"
      >
        Центральный узор
      </a>
    </div>
  </li>
</ul>
`;
